Explain how testing and CI works in more detail

auxesis opened the pull request:
- Add steps describing how to do local development
- Add steps for installing mise
- Add section explaining project layout
- Add explanation of what we're testing, and how changes trigger tests in CI
- Add steps describing how to run tests for different Postgres versions
- Add feedback at the end of `mise run build` that tells the user where to find build artifacts

cisopaul commented on README.md:
```diff
@@ -395,9 +395,14 @@ This will trigger the [Release EQL](https://github.com/cipherstash/encrypt-query
+The easiest way to run the tests is in GitHub Actions:
+
+- [Automatically whenever there are changes](./.github/workflows/test-eql.yml) in the `sql/`, `tests/`, or `tasks/` directories
+- By manually running [the workflow](https://github.com/cipherstash/encrypt-query-language/actions/workflows/test-eql.yml)
```
Is there an arch diagram of what this looks like?

cisopaul replied:
Also ref this comment on line 394 for the release component

cisopaul reviewed (approved):
This looks good for the 'how do I do this?' view, but there is likely more detail needed for the 'what does this system look like & why are certain things true' view. For example what is mise used for & what tests are being run?

Merging this now is ok, but I think we'll need a bit more detail somewhere to match what we have here: https://github.com/cipherstash/cipherstash-suite/blob/main/docs/diagrams/console-api-releases.drawio.svg

auxesis commented:
> This looks good for the 'how do I do this?' view, but there is likely more detail needed for the 'what does this system look like & why are certain things true' view. For example what is mise used for & what tests are being run?

This is good feedback, thanks @cisopaul. 

[I have added docs](https://github.com/cipherstash/encrypt-query-language/pull/81/commits/e47f58d738e1cc4927b5503e6dc0b9ef286801ab) explaining the project layout:

<img width="1405" alt="image" src="https://github.com/user-attachments/assets/a8cf638c-a1d3-42d0-b635-73b3c7a4ccfe" />

[I have added docs](https://github.com/cipherstash/encrypt-query-language/pull/81/commits/b88238c99506c33296f10b0b18c8406e678cfda6) explaining what the tests are testing:

<img width="1297" alt="image" src="https://github.com/user-attachments/assets/92a690cd-d858-4453-b0bf-bc31f7db4ceb" />

[I have added](https://github.com/cipherstash/encrypt-query-language/pull/81/commits/7aca8979cc1d1fd061cc6a71d495c2775a251a2a) a diagram explaining how the testing works:

<img width="1316" alt="image" src="https://github.com/user-attachments/assets/ee972d0c-8bb8-4ecb-bdc5-66d99990e19c" />

cisopaul commented on README.md:
```diff
@@ -393,11 +395,118 @@ To cut a [release](https://github.com/cipherstash/encrypt-query-language/release
+- Automatically whenever there are changes in the `sql/`, `tests/`, or `tasks/` directories
+- By manually running [the workflow](https://github.com/cipherstash/encrypt-query-language/actions/workflows/test-eql.yml)
+
+This is how the `test-eql.yml` workflow functions:
```
This section is really good. We should try to use this in the other places.
